### PR TITLE
feat(core): open in app for self-hosted

### PR DESCRIPTION
**Pull request by pengx17**
<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/T2klNLEk0wxLh4NRDzhk/545994dd-6f7d-468d-a90c-45cb382fdb9d.mp4">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/T2klNLEk0wxLh4NRDzhk/545994dd-6f7d-468d-a90c-45cb382fdb9d.mp4">20241222-1456-24.5006677.mp4</video>

fix AF-1815

### CODE REVIEW

**Comment by pengx17** [comment]
* **#9231** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9231?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9231?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=9231).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by forehalo — APPROVED** [review]

File: packages/frontend/core/src/desktop/pages/workspace/index.tsx [hunk 1]
```diff
@@ -110,6 +117,17 @@ export const Component = (): ReactElement => {
   }, [listLoading, meta, workspaceNotFound, workspacesService]);
 
   if (workspaceNotFound) {
+    if (BUILD_CONFIG.isElectron && searchParams.get('server')) {
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/pages/workspace/index.tsx]:
```suggestion
    if (BUILD_CONFIG.isElectron && serverNotFound) {
```

Comment (EYHN) [reply packages/frontend/core/src/desktop/pages/workspace/index.tsx]: const serversService = useService(ServersService)
const serverNotFound = serversService.getServerByBaseUrl(searchParams.get('server')))

File: packages/frontend/core/src/desktop/pages/workspace/index.tsx [hunk 2]
```diff
@@ -110,6 +117,17 @@ export const Component = (): ReactElement => {
...
+      );
+      return (
+        <AffineOtherPageLayout>
+          <SignIn redirectUrl={redirectUrl} />
+        </AffineOtherPageLayout>
+      );
+    }
     if (detailDocRoute) {
       return (
         <SharePage
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/pages/workspace/index.tsx:141]: <i class='graphite__hidden'>[Re: line +141]<br/><br/></i>

Should consider selfhosted?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9231?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: packages/frontend/core/src/desktop/pages/workspace/index.tsx (hunk 1 above)
Comment (pengx17) [reply packages/frontend/core/src/desktop/pages/workspace/index.tsx]: It is possible that the server already exists but the user has not logged in.

File: packages/frontend/core/src/desktop/pages/workspace/index.tsx (hunk 2 above)
Comment (pengx17) [reply packages/frontend/core/src/desktop/pages/workspace/index.tsx:141]: Server will be stripped off in the second attempt and reach to this point.

File: packages/frontend/core/src/desktop/pages/workspace/index.tsx (hunk 1 above)
Comment (EYHN) [reply packages/frontend/core/src/desktop/pages/workspace/index.tsx]: It also be possible that the user logged in with the wrong account

**Pull request merged**